plugins/modules/storage_box_subaccount.py

```diff
@@ -0,0 +1,468 @@
+from ..module_utils.client import client_resource_not_found
+from ..module_utils.hcloud import AnsibleHCloud, AnsibleModule
+from ..module_utils.storage_box_subaccount import NAME_LABEL_KEY
+from ..module_utils.vendor.hcloud import HCloudException
+from ..module_utils.vendor.hcloud.storage_boxes import (
+    BoundStorageBox,
+    BoundStorageBoxSubaccount,
+    StorageBoxSubaccountAccessSettings,
+)
+
+
+class AnsibleStorageBoxSubaccount(AnsibleHCloud):
+    represent = "storage_box_subaccount"
+
+    storage_box: BoundStorageBox | None = None
+    storage_box_subaccount: BoundStorageBoxSubaccount | None = None
+    storage_box_subaccount_name: str | None = None
+
+    def _prepare_result(self):
+        if self.storage_box_subaccount is None:
+            return {}
+        return storage_box_subaccount.prepare_result(self.storage_box_subaccount, self.storage_box_subaccount_name)
+
+    def _fetch(self):
+        self.storage_box = storage_box.get(self.client.storage_boxes, self.module.params.get("storage_box"))
+
+        if (value := self.module.params.get("id")) is not None:
+            self.storage_box_subaccount = self.storage_box.get_subaccount_by_id(value)
+        elif (value := self.module.params.get("name")) is not None:
+            self.storage_box_subaccount = storage_box_subaccount.get_by_name(self.storage_box, value)
+
+        # Workaround the missing name property
+        # Get the name of the resource from the labels
+        if self.storage_box_subaccount is not None:
+            self.storage_box_subaccount_name = self.storage_box_subaccount.labels.pop(NAME_LABEL_KEY)
+
+    def _create(self):
+        self.fail_on_invalid_params(
+            required=["name", "home_directory", "password"],
+        )
+        params = {
+            "home_directory": self.module.params.get("home_directory"),
+            "password": self.module.params.get("password"),
+        }
+
+        if (value := self.module.params.get("description")) is not None:
+            params["description"] = value
+
+        if (value := self.module.params.get("labels")) is not None:
+            params["labels"] = value
+
+        if (value := self.module.params.get("access_settings")) is not None:
+            params["access_settings"] = StorageBoxSubaccountAccessSettings.from_dict(value)
+
+        # Workaround the missing name property
+        # Save the name of the resource in the labels
+        if "labels" not in params:
+            params["labels"] = {}
+        params["labels"][NAME_LABEL_KEY] = self.module.params.get("name")
+
+        if not self.module.check_mode:
+            resp = self.storage_box.create_subaccount(**params)
+            self.storage_box_subaccount = resp.subaccount
+            self.actions.append(resp.action)
+
+            self._wait_actions()
+            self.storage_box_subaccount.reload()
+            self.storage_box_subaccount_name = self.storage_box_subaccount.labels.pop(NAME_LABEL_KEY)
```

Comment:
Value is only read in `_update`, so this write is never used.

Reply:
We must always `pop` (get and remove) the name from the labels, so that `prepare_result` does not return the name label to the user.

Every time we fetch the resource from the API, the above must happen.

Reply:
Ack